lib/utils.ts

```diff
@@ -50,14 +50,14 @@ export const getAppConfig = cache(async (headers: Headers): Promise<AppConfig> =
       });
 
       const remoteConfig: SandboxConfig = await response.json();
-      const config: AppConfig = { ...APP_CONFIG_DEFAULTS };
+      const config: AppConfig = { sandboxId, ...APP_CONFIG_DEFAULTS };
 
       for (const [key, entry] of Object.entries(remoteConfig)) {
         if (entry === null) continue;
         if (
-          key in config &&
-          typeof config[key as keyof AppConfig] === entry.type &&
-          typeof config[key as keyof AppConfig] === typeof entry.value
+          (key in config && config[key as keyof AppConfig] === undefined) ||
```

Comment:
this logic was hard for me to grok until it clicked

maybe we could add a comment to make it clearer the intention

which I understand as, 
we want to set the fetched value if the key is defined on `APP_CONFIG_DEFAULTS` but the value is `undefined`

would this make it more explicit? (I'm not sure)
```suggestion
  (key in APP_CONFIG_DEFAULTS && APP_CONFIG_DEFAULTS[key as keyof AppConfig] === undefined) ||
```

Reply:
Yes!